universal-application-tool-0.0.1/test/repository/QuestionRepositoryTest.java

```diff
@@ -69,100 +68,61 @@ public void findPathConflictingQuestion_noConflicts_ok() throws Exception {
             .setPath(Path.create("applicant.a_brand_new_question"))
             .build();
 
-    Optional<Question> maybeConflict = repo.findPathConflictingQuestion(newQuestionDefinition);
+    Optional<Question> maybeConflict = repo.findConflictingQuestion(newQuestionDefinition);
 
     assertThat(maybeConflict).isEmpty();
   }
 
   @Test
-  public void findPathConflictingQuestion_sameQuestion_hasConflict() {
+  public void findConflictingQuestion_sameQuestionPathSegment_hasConflict() throws Exception {
     Question applicantAddress = testQuestionBank.applicantAddress();
-    Optional<Question> maybeConflict =
-        repo.findPathConflictingQuestion(applicantAddress.getQuestionDefinition());
-
-    assertThat(maybeConflict).contains(applicantAddress);
-  }
-
-  @Test
-  public void findPathConflictingQuestion_differentVersion_hasConflict() throws Exception {
-    Question applicantName = testQuestionBank.applicantName();
-    QuestionDefinition questionDefinition =
-        new QuestionDefinitionBuilder(applicantName.getQuestionDefinition()).setId(123123L).build();
-
-    Optional<Question> maybeConflict = repo.findPathConflictingQuestion(questionDefinition);
-
-    assertThat(maybeConflict).contains(applicantName);
-  }
-
-  @Test
-  public void findPathConflictingQuestion_samePath_hasConflict() throws Exception {
-    Question applicantName = testQuestionBank.applicantName();
-    QuestionDefinition questionDefinition =
-        new QuestionDefinitionBuilder(applicantName.getQuestionDefinition()).clearId().build();
-
-    Optional<Question> maybeConflict = repo.findPathConflictingQuestion(questionDefinition);
-
-    assertThat(maybeConflict).hasValue(applicantName);
-  }
-
-  @Test
-  public void findPathConflictingQuestion_repeaterConflictsWithNonRepeater_hasConflict()
-      throws Exception {
-    Question householdMembers = testQuestionBank.applicantHouseholdMembers();
     QuestionDefinition newQuestionDefinition =
-        new QuestionDefinitionBuilder(householdMembers.getQuestionDefinition())
+        new QuestionDefinitionBuilder(applicantAddress.getQuestionDefinition())
             .clearId()
-            .setPath(householdMembers.getQuestionDefinition().getPath().withoutArrayReference())
-            .setQuestionType(QuestionType.TEXT)
+            .setName("applicant address!")
+            .setPath(Path.create("fake"))
```

Comment:
Question (non-blocking): is this necessary or used at all?

Reply:
are you talking about this test case, or `setPath(Path.create("fake"))`?

The fake path is not useful. Hopefully very soon we can delete all paths from questions, including the `QuestionDefinitionBuilder#setPath` and so many other things. But for now, I think the path needs to be set or the builder won't work.

Reply:
Ah I see. Thanks!